OlinkAnalyze/R/read_flex.R

```diff
@@ -0,0 +1,201 @@
+    LOD <- dat |> dplyr::filter(stringr::str_detect(SampleID, "Plate LOD")) |> 
+      tidyr::pivot_longer(cols = tidyselect::starts_with("OID"),
+                          names_to = "OlinkID",
+                          values_to = "Plat_LOD") |> 
+      dplyr::select(-tidyselect::starts_with("QC")) |> 
+      dplyr::select(-SampleID)
+    
+    LLOQ <- dat |> dplyr::filter(stringr::str_detect(SampleID, "LLOQ"))  |> 
+      tidyr::pivot_longer(cols = tidyselect::starts_with("OID"),
+                          names_to = "OlinkID",
+                          values_to = "LLOQ") |> 
+      dplyr::select(-SampleID, -tidyselect::starts_with("PlateID"), -tidyselect::starts_with("QC"))
+    
+    ULOQ <- dat |> dplyr::filter(stringr::str_detect(SampleID, "ULOQ"))  |> 
+      tidyr::pivot_longer(cols = tidyselect::starts_with("OID"),
+                          names_to = "OlinkID",
+                          values_to = "ULOQ") |> 
+      dplyr::select(-SampleID, -tidyselect::starts_with("PlateID"), -tidyselect::starts_with("QC"))
+    
+    meta_data_per_assay <- dplyr::left_join(missfreq,norm_method, by = "OlinkID") |> 
+      dplyr::left_join(assay_warning, by = "OlinkID", multiple ="all") |>
+      dplyr::left_join(LLOQ, by = c("OlinkID")) |> 
+      dplyr::left_join(ULOQ, by = c("OlinkID")) |> 
+      dplyr::left_join(Plate_LQL, by = c("OlinkID", "PlateID1")) |> 
+      dplyr::left_join(LOD, by = c("OlinkID", "PlateID1")) 
+  } else {
+    LOD <- dat |> dplyr::filter(stringr::str_detect(SampleID, "LOD")) |> 
+      dplyr::select(-SampleID, -tidyselect::starts_with("PlateID"), -tidyselect::starts_with("QC")) |> 
+      tidyr::pivot_longer(cols = tidyselect::starts_with("OID"),
+                          names_to = "OlinkID",
+                          values_to = "LOD") 
+    meta_data_per_assay <- dplyr::left_join(missfreq,norm_method, by = c("OlinkID")) |> 
+      dplyr::left_join(LOD, by = c("OlinkID")) |> 
+      dplyr::select(-tidyselect::starts_with("PlateID"), -tidyselect::starts_with("QC"))
+  }
+  
+  
+  # pivot longer
+  dat<- dat |> 
+    dplyr::filter(!str_detect(SampleID, paste( "Missing Data freq.",
+                                               "Normalization",
+                                               "Assay warning",
+                                               "Lowest quantifiable level",
+                                               "Plate LOD",
+                                               "LLOQ",
+                                               "ULOQ",
+                                               "LOD", 
+                                               collapse = "|"))) |>  # remove end meta data
+    tidyr::pivot_longer(cols = tidyselect::starts_with("OID"),
+                        names_to = "OlinkID",
+                        values_to = ifelse(is_npx_data, "NPX", "Quantified_value")) |> 
+    dplyr::left_join(metadata, by = c("OlinkID")) |> 
+    dplyr::left_join(meta_data_per_assay, multiple = "all") |> 
```

Comment:
This line 
"dplyr::left_join(meta_data_per_assay, multiple = "all")" 
will print out "Joining with `by = join_by(OlinkID)`" which might be confusing when reading in data.